module-apps/apps-common/widgets/BellSideListItem.cpp

```diff
@@ -10,9 +10,7 @@ namespace gui
 {
     BellSideListItem::BellSideListItem(const std::string &description) : ListItemWithDescription(description)
     {
-        setMinimumSize(style::sidelistview::list_item::w, style::sidelistview::list_item::h);
-        setEdges(RectangleEdge::None);
-        body = new BellBaseLayout(this);
+        init();
```

Comment:
This is more of a side note because AFAIU this code was here before this PR but I'd like to point out a few things.

There are two `new` calls in the constructor. If an exception is thrown by any other call, e.g. by the second `new` then the allocated memory will not be freed.

Who owns `topMessage`? It's a local variable. The memory is allocated and is not freed anywhere. The same actually applies for `body` and `bottomText`. Am I missing something?

What is the reason of not using smart pointers? Or even better would it be possible to avoid pointers altogether and store these objects as members of this item?

The calls in this constructor are more or less the same as the ones in `setupBottomDescription()`. Shouldn't this logic be encapsulated in some common function called in both places?

Reply:
**topMessage** and **bottomMessage** are owned by **body** , **body** is owned by **this**, as we discussed internally there is not memory leakage. The reason why we cant use smart pointers was already mentioned earlier _"Generally gui design is based on raw pointers and memory is managed manually. The destructor is called in base/parent/Item class"_

Reply:
Thanks for the explanation. Note that the memory is going to be leaked here if an exception is thrown but currently the whole system will crash if that happens so memory leak is less important I guess.